### PR TITLE
[test] Create basic engine integration tests

### PR DESCRIPTION
**Pull request by glanham-jr**
## What does this PR do?

Create a new integration unit test. This test allows testing for all engines, or just a single engine, that is currently configured in the `settings.yml` file.

## Why is this change important?

When developing the Mullvad Leta integration, I realized every change I would make would result in me going back to the local host running (albeit auto deployed) and testing the results, then going back and analyzing the log. I realized I could speed this up if I had a super quick command line way to do so. Now, if I want to make changes to an engine, I can just configure the `.env.test` file with my engine, run `make test.int`, and I get immediate results for a tighter feedback loop.

## How to test this PR locally?

1. run `make test.int` to run the test for all engines
2. To test a specific engine, configure `.env.test` and set the `INTEGRATION_TEST_ENGINES` equal to whatever engines(s) you wish to test.

## Author's checklist

- [ ] Need to add documentation on this test
- [ ] Address feedback

## Related issues
n/a

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -21,3 +21,4 @@ aiounittest==1.4.2
 yamllint==1.33.0
 wlc==1.13
 coloredlogs==15.0.1
+python-dotenv==1.0.0
```
Comment (glanham-jr) [inline requirements-dev.txt:24]: note I needed to install dotenv to load the environment variables easily. I could parse it manually, but this seemed better.

File: Makefile [hunk 2]
```diff
@@ -84,7 +84,7 @@ MANAGE += py.build py.clean
 MANAGE += pyenv pyenv.install pyenv.uninstall
 MANAGE += pypi.upload pypi.upload.test
 MANAGE += format.python
-MANAGE += test.yamllint test.pylint test.pyright test.black test.pybabel test.unit test.coverage test.robot test.rst test.clean
+MANAGE += test.yamllint test.pylint test.pyright test.black test.pybabel test.unit test.coverage test.robot test.rst test.int test.clean
```
Comment (glanham-jr) [inline Makefile:87]: Not sure if this is needed, will need help determining this

**Comment by glanham-jr** [comment]
I realize that some of the files got re-formatted - if the general concept is OK, I can undo the styling.

File: tests/integration/test_engines.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+logger.addHandler(stream_handler)
+
+
+def test_single_engine(app: Flask, engine_name: str) -> Tuple[str, Optional[Exception], int]:
+    logger.debug(f'Testing Engine: {engine_name}')
+    try:
+        with app.test_request_context():
+            # test your app context code
+            search_query = SearchQuery(
+                'test', [EngineRef(engine_name, 'general')], 'en-US', SAFESEARCH, PAGENO, None, None
```
Comment (glanham-jr) [inline tests/integration/test_engines.py:29]: can configure the search term in an environment variable - can do so later if PR concept is OK

**Comment by dalf** [comment]
Is the embedded checker related to this PR ?

https://github.com/searx/searx/pull/2419
https://docs.searxng.org/dev/makefile.html#make-search-checker-engine-name

**Comment by glanham-jr** [comment]
Hmmmm, that may be the exact same functionality this is, but I wasn't aware of it. I'll check it out later.

**Comment by return42** [comment]
@glanham-jr what is the state of this PR .. you are aware this PR is still in DRAFT?

**Pull request closed without merging**